src/main/java/core/basesyntax/dao/animal/AnimalDaoImpl.java

```diff
@@ -12,11 +15,39 @@ public AnimalDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Animal save(Animal animal) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(animal);
+            transaction.commit();
+            return animal;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert animal entity "
```

Comment:
```suggestion
            throw new RuntimeException("Can't insert animal entity " + animal, e);
```
we can inline if checkstyle allows